crates/ide/src/folding_ranges.rs

```diff
@@ -264,6 +270,24 @@ fn fold_range_for_where_clause(where_clause: ast::WhereClause) -> Option<TextRan
     None
 }
 
+fn fold_range_for_multiline_match_arm(match_arm: ast::MatchArm) -> Option<TextRange> {
+    match match_arm.expr()?.syntax().kind() {
+        BLOCK_EXPR | MATCH_EXPR | ARRAY_EXPR | RECORD_EXPR => {
+            return None;
+        }
+        _ => {}
+    }
+    if let Some(newline_offset) = match_arm.syntax().text().find_char('\n') {
```

Comment:
```suggestion
    if match_arm.expr()?.syntax().text().contains('\n') {
```
I don't think we want to do this when the match arm is multiline due to its pattern, so we should just check the expression. Would be good to also include a match arm with a multiline pattern in a test to check that.